socs/agents/hwp_gripper/agent.py

```diff
@@ -460,6 +515,10 @@ def run_and_append(func, *args, **kwargs):
 
                     finished[actuator] = True
 
+        if (not all(finished)) or any(aborted):
+            self.log.error('HWP is not correctly gripped')
+            run_and_append(self.client.home, job='grip', check_shutdown=check_shutdown)
```

Comment:
For this to work consistently, you should probably turn on the `force` function before moving home